[ethan-goh] iP

ethan-goh opened the pull request:
### **Genesis**

> "Efficiency is doing things right; effectiveness is doing the right things." - Peter Drucker ([source](https://businessfitness.biz/efficiency-effectiveness-right-things-right/))

Welcome to Genesis, your **very own task tracker**! With Genesis, you can:

- Create your own tasks ✍️
- Mark tasks as done ✅
- Delete tasks that are no longer needed ❌

Tasks that are supported include:

1. Deadlines
2. Events
3. Todos

Genesis is

- [ ] text-based
- [ ] easy to learn
- [ ] SUPER FAST to use

If you are a Java programmer, you can use it to practice Java too. Here's the ```main``` method:
``` java
 public static void main(String[] args) {
        System.out.println("Hello! I'm Genesis!\nWhat can I do for you?\n");

        TaskManager taskManager = new TaskManager();
        Storage storage = new Storage(taskManager);
        CommandParser parser = new CommandParser(taskManager, storage);
        Ui ui = new Ui(taskManager, parser);
        storage.loadTasks(parser, ui);


        Scanner sc = new Scanner(System.in);
        while (true) {
            String input = sc.nextLine();
            ui.handleInput(input, false);
        }
    }
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

ethan-goh committed:
completed level 0

ethan-goh committed:
level 1

ethan-goh committed:
completed level 2

ethan-goh committed:
completed level 3

ethan-goh committed:
finished level 4

ethan-goh committed:
finished TextUITesting

ethan-goh committed:
finished level 5

ethan-goh committed:
completed Level 6

ethan-goh committed:
Reformat code design to include Task Manager and Command Parser classes in order to compartmentalises certain functionalities, encapsulate task-related operations.

ethan-goh committed:
Implement reading and writing data files for Level-7

ethan-goh committed:
Merge branch 'branch-Level-7'
Merge Level 7 branch with master branch

ethan-goh committed:
Implement reading of date time in YYYY-MM-DD format and printing in different for format  for Level-8

ethan-goh committed:
Merge branch 'branch-Level-8'
Merge branch 'branch-Level-8
'

ethan-goh committed:
Add Ui and Storage classes while allowing interaction between classes to extract out closely related code as classes

ethan-goh committed:
Put all classes into one package Genesis

ethan-goh committed:
Merge remote-tracking branch 'origin/add-gradle-support'

Merge gradle support

ethan-goh committed:
Add simple test cases for taskmanager and configured JUnit

ethan-goh committed:
Add JavaDoc comments

ethan-goh committed:
Edit coding standard

ethan-goh committed:
Add functionality to deal with find input

ethan-goh committed:
Merge branch 'branch-Level-9'
Merge level-9 branch with master branch

ethan-goh committed:
Add JAR Files

vinceg4 commented on src/main/java/genesis/Genesis.java:
```diff
@@ -0,0 +1,530 @@
+ */
+
+
+class Task {
```
Would it be better to separate the different classes into different files instead?

vinceg4 commented on src/main/java/genesis/Genesis.java:
```diff
@@ -0,0 +1,530 @@
+/**
+ * Represents a generic task.
+ */
+
+
```
Should these blank lines be removed to follow the coding standard?

vinceg4 commented on src/main/java/genesis/Genesis.java:
```diff
@@ -0,0 +1,530 @@
+import java.io.IOException;
+import java.util.Scanner;
+import java.util.ArrayList;
+import java.io.File;
```
Should the java.io imports be placed together to keep the format of import order consistent?

vinceg4 commented on src/main/java/genesis/Genesis.java:
```diff
@@ -0,0 +1,530 @@
+    /**
+     * Writes the current list of tasks to a file.
+     */
+    public void writeTasks() {
```
I like how simple your` writeTasks` function is, I think it is intuitive and easy to understand 👍🏻

ethan-goh committed:
Add Checkstyle config

ethan-goh committed:
Merge branch-A-CheckStyle into master

ethan-goh committed:
Integrate JavaFX to create a GUI

ylchin reviewed (commented):
good naming of variables, and javadocs added for every file, but might want to check your blank lines across the different files. according to the java coding standards, "No blank line between the documentation block and the method/class."

ylchin commented on src/main/java/commandparser/CommandParser.java:
```diff
@@ -0,0 +1,166 @@
+package commandparser;
+
+import storage.Storage;
+import taskmanager.TaskManager;
+import task.Task;
+import deadline.Deadline;
+import todo.Todo;
+import event.Event;
+
+
+import java.time.LocalDate;
```
unnecessary blank line, import java should come before other imports

ylchin commented on src/main/java/commandparser/CommandParser.java:
```diff
@@ -0,0 +1,166 @@
+public class CommandParser {
+    protected TaskManager taskManager;
+    protected Storage storage;
+    /**
```
insert blank line before javadoc

ylchin commented on src/main/java/commandparser/CommandParser.java:
```diff
@@ -0,0 +1,166 @@
+     * @param input The user input containing the task description, start, and end time.
+     * @param isSilent If true, suppress output messages.
+     */
+    public String handleEvent(String input, boolean isSilent) {
+        if (!input.contains("/from ") || !input.contains("/to ")) {
+            return "You need a starting and ending date to add this task!";
+
+        }
+        String[] parts = input.split("/from ");
+        String[] dateParts = parts[1].split("/to ");
+        String taskName = parts[0].replaceFirst("event ", "").trim();
+        if (taskName.isEmpty()) {
+            return "You need a task description!";
+
+        }
+        String initialStartDate = dateParts[0].trim();
+        String initialEndDate = dateParts[1].trim();
+        LocalDate startDate = LocalDate.parse(initialStartDate);
+        LocalDate endDate = LocalDate.parse(initialEndDate);
+        Event eventTask = new Event(taskName, startDate, endDate, input);
+        String temp = taskManager.addTask(eventTask, isSilent);
+        this.storage.writeTasks();
+        return temp;
+    }
+}
```
insert blankline after if block and use blank line to separate different logic blocks

ylchin commented on src/main/java/deadline/Deadline.java:
```diff
@@ -0,0 +1,34 @@
+     */
+    @Override
+    public String toString() {
+        return "[D]" + super.toString() + " (by: " + this.deadline.format(DateTimeFormatter.ofPattern("MMM d yyyy")) + ")";
```
line exceeds 120 character limit

ylchin commented on src/main/java/genesis/Genesis.java:
```diff
@@ -0,0 +1,77 @@
+ */
+
+
+
+
+
+
+
+/**
+ * Manages a list of tasks.
+ */
+
+/**
+ * Handles loading tasks from and writing tasks to a file.
+ */
+
+
+
+
+
+
```
Why do you need more than 1 javadoc for a class?

ylchin commented on src/main/java/main/Main.java:
```diff
@@ -0,0 +1,32 @@
+package main;
+import java.io.IOException;
```
insert blankline before import

ylchin commented on src/test/java/genesis/GenesisTest.java:
```diff
@@ -0,0 +1,30 @@
+package genesis;
+import org.junit.jupiter.api.Test;
```
insert blankline before import

ylchin reviewed (commented):
Good naming of variables, might want to be more mindful of import orders, sticking to one class header and blank spaces

ethan-goh committed:
Add assertions in TaskManager class

ethan-goh committed:
Merge pull request #1 from ethan-goh/branch-A-Assertions

Add assertions in TaskManager class

ethan-goh committed:
refactor: Improve code readability by simplifying expressions

The current code contains a complex expression that reduces clarity
and makes it harder for developers to follow the logic. The happy
path in one section is not immediately apparent, causing potential
confusion.

This commit simplifies the complicated expression to enhance code
readability and makes the happy path more prominent. This change
improves maintainability and allows developers to understand the
intended flow of the code more easily.

By refactoring in this way, we adhere to clean code principles,
reducing cognitive load and making future modifications easier.

ethan-goh committed:
Merge pull request #2 from ethan-goh/branch-A-CodeQuality

refactor: Improve code readability by simplifying expressions

ethan-goh committed:
Use Java Streams for task listing in TaskManager

The current `listTasks` method uses a traditional for-loop to generate
the list of tasks, which can be made more concise and readable using
Java Streams.

This commit refactors the `listTasks` method to utilize Java Streams
for generating the numbered list of tasks. By using the Stream API,
we improve code readability and maintainability, leveraging functional
programming concepts for a more declarative approach.

The change helps:
- Reduce boilerplate code by using `map` to format the tasks with their
  indices.
- Utilize `Collectors.joining("\n")` to construct the output efficiently.

Adopting Streams enhances code clarity and aligns with modern Java
practices.

ethan-goh committed:
Merge pull request #3 from ethan-goh/branch-A-Streams

Use Java Streams for task listing in TaskManager

ethan-goh committed:
Add extension to check for duplicate tasks when adding

ethan-goh committed:
Merge remote-tracking branch 'origin/master'

ethan-goh committed:
Add user guide and Ui.png

ethan-goh committed:
Utilise exception more
